iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -493,20 +501,47 @@ private String getStatsSource() {
   }
 
   private Path getStatsPath(Table table) {
-    return new Path(table.location() + STATS + table.name() + table.currentSnapshot().snapshotId());
+    return getStatsPath(table, table.currentSnapshot().snapshotId());
+  }
+
+  private Path getStatsPath(Table table, long snapshotId) {
+    return new Path(table.location() + STATS + table.name() + snapshotId);
   }
 
-  private void invalidateStats(Path statsPath) {
+  private void checkAndMergeOrInvalidateStats(ColumnStatistics statsObjNew,
+      org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
+    Table tbl = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
+    Path statsPath = getStatsPath(tbl);
     try {
       FileSystem fs = statsPath.getFileSystem(conf);
       if (fs.exists(statsPath)) {
+        // Analyze table and stats updater thread
         fs.delete(statsPath, true);
+      } else {
+        // Insert queries
+        mergeStats(statsObjNew, tbl, hmsTable);
       }
     } catch (IOException e) {
       LOG.error("Failed to invalidate stale column stats: {}", e.getMessage());
     }
   }
 
+  private void mergeStats(ColumnStatistics statsObjNew, Table tbl, org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
+    if (tbl.currentSnapshot().parentId() != null) {
+      long previousSnapshotId = tbl.currentSnapshot().parentId();
```

Comment:
move it 1 line up and check var in the if block

Reply:
done